Chess/movevalidator.cpp

```diff
@@ -0,0 +1,92 @@
+#include "movevalidator.h"
+
+
+MoveValidator::MoveValidator(){}
+
+
+bool MoveValidator::checkMove(int diffRow, int diffCol, int row, int column)
+{
+    const std::vector<std::pair<int,int>> moves = {{1, 2}, {1, -2}, {-1, -2}, {-2, 1},
+                                           {2, -1}, {2, 1}, {-1, 2}, {-2, -1}};
+
+    bool isHorse = false;
+
+    for(auto j : moves)
+        isHorse |= (diffCol == j.second && diffRow == j.first);
+
+    if(isHorse)
+        return true;
+
+    int moveX = 0, moveY = 0;
+    if(diffRow)
+        moveX = diffRow / std::abs(diffRow);
+    if(diffCol)
+        moveY = diffCol / std::abs(diffCol);
+
+    diffRow = std::abs(diffRow), diffCol = std::abs(diffCol);
+
+
+    while(diffRow > 0 || diffCol > 0)
+    {
+        row += moveX;
+        column += moveY;
+        diffRow--, diffCol--;
+
+        int type = checkCell(row, column);
+
+        if(type == 1)
+            return 0;
+
+        if(type == 0)
+        {
+            if(std::max(diffRow, 0) + std::max(diffCol, 0) == 0)
+                return 1;
+            else
+                return 0;
+        }
+    }
+    return 1;
+}
+
+int MoveValidator::checkCell(int row, int column)
```

Comment:
as redenumi in validateCell si as face tot aici verificarile de deasupra intre liniile 35-45